pkg/systemd/shutdown.jsx

```diff
@@ -0,0 +1,198 @@
+            const input_timestamp = parseInt(data, 10);
+            const server_timestamp = parseInt(this.server_time.now.getTime() / 1000, 10);
+            let offset = Math.ceil((input_timestamp - server_timestamp) / 60);
+
+            /* If the time in minutes just changed, make it happen now */
+            if (offset === -1) {
+                offset = 0;
+            } else if (offset < 0) { // Otherwise it is a failure
+                this.setState({ dateError: _("Cannot schedule event in the past") });
+                return;
+            }
+
+            this.setState({
+                when: "+" + offset,
+                error: "",
+                dateError: "",
+            });
+        })
+                .catch(e => this.setState({ error: e }));
+    }
+
+    onSubmit() {
+        const arg = this.props.shutdown ? "--poweroff" : "--reboot";
+        if (!this.props.shutdown)
+            cockpit.hint("restart");
+
+        cockpit.spawn(["shutdown", arg, this.state.when, this.state.message], { superuser: true, err: "message" })
+                .then(this.props.onClose)
+                .catch(e => this.setState({ error: e }));
+    }
+
+    render() {
+        const options = [
+            <SelectOption value="1" key="1">{_("1 minute")}</SelectOption>,
+            <SelectOption value="5" key="5">{_("5 minutes")}</SelectOption>,
+            <SelectOption value="20" key="20">{_("20 minutes")}</SelectOption>,
+            <SelectOption value="40" key="40">{_("40 minutes")}</SelectOption>,
+            <SelectOption value="60" key="60">{_("60 minutes")}</SelectOption>,
+            <Divider key="divider" component="li" />,
+            <SelectOption value="0" key="0">{_("No delay")}</SelectOption>,
+            <SelectOption value="x" key="x">{_("Specific time")}</SelectOption>
+        ];
+
+        return (
+            <Modal isOpen position="top" variant="medium"
+                   onClose={this.props.onClose}
+                   id="shutdown-dialog"
+                   title={this.props.shutdown ? _("Shut down") : _("Restart")}
+                   footer={<>
+                       <Button variant='danger' isDisabled={this.state.error || this.state.dateError} onClick={this.onSubmit}>{this.props.shutdown ? _("Shut down") : _("Restart")}</Button>
+                       <Button variant='link' onClick={this.props.onClose}>{_("Cancel")}</Button>
+                   </>}
+            >
+                <>
+                    <Form isHorizontal onSubmit={e => {
+                        // HACK: https://github.com/patternfly/patternfly-react/issues/5299
+                        e.preventDefault();
+                        return false;
+                    }}>
+                        <FormGroup fieldId="message" label={_("Message to logged in users")}>
+                            <TextArea id="message" value={this.state.message} onChange={v => this.setState({ message: v })} />
```

Comment:
Text area allows you to expand it horizontaly and vertically. When inside modals vertical works fine, the modal get's expanded as well, but the horizontal escapes outside of the modal. I have not reported it yet to PF, I need to check if it's related to any of our custom CSS.

Reply:
I've added `resize:vertical` so it can be resized only vertically.